src/middleware/imageUpload.js

```diff
@@ -15,31 +15,36 @@ const uploadToCloudinary = async (filePath) => {
     let mainFolderName = "tusomeniImages"
     let filePathOnCloudinary = `${mainFolderName}/${path.basename(filePath)}`
 
-    return cloudinary.uploader.upload(filePath, {"public_id": filePathOnCloudinary})
-    .then(res => {
-        fs.unlinkSync(filePath) // remove file from local upload folder
-
-        return {
-            message: "Success",
-            url: res.url
-        }
-    }).catch(err => {
-        fs.unlinkSync(filePath);
-        throw new Error("Failed to upload images")
-    })
+    return cloudinary.uploader.upload(filePath, { "public_id": filePathOnCloudinary })
+        .then(res => {
+            fs.unlinkSync(filePath) // remove file from local upload folder
+
+            return {
+                message: "Success",
+                url: res.url
+            }
+        }).catch(err => {
+            fs.unlinkSync(filePath);
+            throw new Error("Failed to upload images")
+        })
 
 }
 
 
 const storage = multer.diskStorage({
     destination: function (req, file, cb) {
-      cb(null, './uploads')
+        cb(null, '')
     },
     filename: function (req, file, cb) {
-      cb(null, file.originalname)
-    }
+        cb(
+            null,
+            `${Math.floor(
+                100000 + Math.random() * 900000
```

Comment:
In retrospect, we could probably use better file names so we can have a structure for the image storage. That way, it will be easier to tell what a paper is for. 

something like "discrete-math-89073.png" instead of "21474647.png"

Reply:
@Asahel-code are you in a position to change this implementation?

Reply:
Yes, I will try and come up with a solution to that